### PR TITLE
Modify random_get to use crypto.getRandomValues

### PR DESCRIPTION
**Pull request by Aandreba**
I simply modified the `random_get` implementation to use `crypto.getRandomValues` to generate the random numbers.
This has the benefit that the RNG is cryptographically secure, and is probably faster that the previous implementation (though that should be benchmarked).

### CODE REVIEW

File: src/wasi.ts [hunk 1]
```diff
@@ -826,8 +827,21 @@ export default class WASI {
...
       random_get(buf: number, buf_len: number) {
         const buffer8 = new Uint8Array(self.inst.exports.memory.buffer);
-        for (let i = 0; i < buf_len; i++) {
-          buffer8[buf + i] = (Math.random() * 256) | 0;
+        const end = buf + buf_len;
+
+        if (self.hasCrypto) {
+          let i = buf;
+          for (; i < end; ) {
+            const next_i = i + 65_536;
```
Comment (bjorn3) [inline src/wasi.ts:835]:
```suggestion
          let i = buf;
          for (let i = buf; i < end; ) {
            const next_i = i + 65_536;
```

File: src/wasi.ts [hunk 2]
```diff
@@ -826,8 +827,21 @@ export default class WASI {
       sched_yield() {},
       random_get(buf: number, buf_len: number) {
         const buffer8 = new Uint8Array(self.inst.exports.memory.buffer);
-        for (let i = 0; i < buf_len; i++) {
-          buffer8[buf + i] = (Math.random() * 256) | 0;
+        const end = buf + buf_len;
+
+        if (self.hasCrypto) {
```
Comment (bjorn3) [inline src/wasi.ts:832]: Can you check `"crypto" in globalThis` instead and not cache it in an instance variable?

Comment (Aandreba) [reply src/wasi.ts:832]: I tried, but it didn't pass the test. Looks like `globalThis` doesn't exist on all environments :disappointed:

File: src/wasi.ts [hunk 3]
```diff
@@ -826,8 +827,21 @@ export default class WASI {
...
+          let i = buf;
+          for (; i < end; ) {
+            const next_i = i + 65_536;
+            crypto.getRandomValues(
+              buffer8.subarray(i, next_i > end ? end : next_i),
+            );
+            i = next_i;
+          }
```
[the remark this answers is not in the document]
Comment (Aandreba) [reply src/wasi.ts:840]: True, I'll refactor it

File: src/wasi.ts (hunk 2 above)
Comment (Aandreba) [reply src/wasi.ts:833]: Also, caching it is most probably faster than checking every time `random_get` gets called

Comment (bjorn3) [reply src/wasi.ts:833]: Did it fail in node.js? Maybe the node.js version that was used is too old? `globalThis` is supposed to work in all environments. It was introduced as way to unify `window`, `self` and `global` across the various javascript environments.

As for performance, if `random_get` is called rarely a tiny slowdown shouldn't matter and if it is called frequently, I would expect the JIT to optimize it down to a couple instructions max, which is much less than even the overhead of making a native call, let alone actually running the PRNG.

Comment (Aandreba) [reply src/wasi.ts]: I remembered it failing on some test, but it seems I was mistaken. I've pushed the uncached `"crypto" in globalThis` version.

**Comment by bjorn3** [comment]
Thanks!

**Pull request merged**